### ISSUE TITLE
python3 compatibility?

### ISSUE THREAD

**Issue by mathog** [Electrostatics/apbs#253]
Working on CentOS 8, which has "python2" and "python3" but not "python".  Installing apbs was mostly OK, but pdb2pqr much less so, and the interaction between the two is not working at all well.  Some of this is probably because python 2 was EOL'd on 1/12020 but apbs/pdb2pqr still uses it.   Likely there is a "python" in there somewhere where only python2 or python3 is available.

But scons.py does not run with python3.  
```
python3 scons/scons.py
  File "scons/scons.py", line 194
    print "  %s" % path
               ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("  %s" % path)?

```

apbs was built with (note, "extract" is from "drm_tools" on sourceforge, the here it is just doing substitions: rtss string2 replaces rtds string1.)

```
pversion=1.5 #determined after it was built
package=apbs
TOPDIR=/usr/common/modules/el8/x86_64/software/${package}/${pversion}-CentOS-vanilla
THELUA=/usr/common/modules/el8/x86_64/modules/all/${package}/${pversion}-CentOS-vanilla.lua
cp ../src/routines.c ../src/routines.c.dist
extract -in ../src/routines.c.dist -out ../src/routines.c \
  -rtds 'for (int i = 0;' -rtss 'int i; for(i = 0;'
cp ../CMakeLists.txt ../CMakeLists.txt.dist
extract -in ../CMakeLists.txt.dist -out ../CMakeLists.txt \
  -rtds '_INSTALL_PATH ' -rtss "_INSTALL_PATH $TOPDIR/"
git clone https://github.com/Electrostatics/apbs-pdb2pqr.git
cd apbs-pdb2pqr
git submodule init
git submodule update
module load msms
module load nanoshaper
module load cmake #very recent release needed
#Requires: OpenBLAS, installed on all nodes
mkdir -p apbs/build
cd apbs/build
cmake -DENABLE_GEOFLOW=ON -DENABLE_PBAM=ON \
   -DENABLE_BEM=ON -DENABLE_FETK=ON \
   -DENABLE_PYTHON=ON -DBUILD_SHARED_LIBS=ON \
   -DGET_NanoShaper=ON ..
make 2>&1 | tee build_2020_04_21.log
#that worked!
make install 2>&1 | tee install_2020_04_21.log
#does NOT copy all the libraries
cp bem-prefix/src/bem-build/libTABIPBlib.so $TOPDIR/lib
(cd fetk/src/fetk-punc-build/src/aaa_lib/.libs; tar -cf - .) | (cd $TOPDIR/lib; tar -xf -)
(cd fetk/src/fetk-mc-build/src/aaa_lib/.libs; tar -cf - .) | (cd $TOPDIR/lib; tar -xf -)
(cd fetk/src/fetk-maloc-build/src/aaa_lib/.libs; tar -cf - .) | (cd $TOPDIR/lib; tar -xf -)
(cd fetk/src/fetk-gamer-build/src/aaa_lib/.libs; tar -cf - .) | (cd $TOPDIR/lib; tar -xf -)
/bin/rm $TOPDIR/lib/*.la
(cd fetk/lib; tar -cf - *.la) | (cd $TOPDIR/lib; tar -xf -)
module_generate_from_directory.sh \
   $package \
   $pversion \
   CentOS/vanilla \
   $TOPDIR \
   "Solve the equations of continuum electrostatics for large biomolecular assemblages." \
   "https://github.com/Electrostatics/apbs-pdb2pqr"

```
however, pdb2pqr downloaded that way would not build because html/server.html.in etc. are NOT in the download,  So obtained and built pdb2pqr like this:

```
module load apbs
pversion=2.1.1
package=pdb2pqr
TOPDIR=/usr/common/modules/el8/x86_64/software/${package}/${pversion}-CentOS-vanilla
THELUA=/usr/common/modules/el8/x86_64/modules/all/${package}/${pversion}-CentOS-vanilla.lua
cd /usr/common/src
wget https://github.com/Electrostatics/${package}/archive/v${pversion}.tar.gz
gunzip -c v${pversion}.tar.gz | tar -xf - 
/bin/rm v${pversion}.tar.gz
cd ${package}-${pversion}
cp build_config.py build_config.py.dist
echo "" >> build_config.py.dist #fix missing newline
#NOTE: next also converts from \r\n to \n!  diff will show all lines!
extract -in build_config.py.dist -out build_config.py \
   -if '#PREFIX="~/pdb2pqr"' -fmt "PREFIX=\"$TOPDIR\""
#needed python2-devel RPM
python2 scons/scons.py 2>&1 | tee build_2020_07_28.log
python2 scons/scons.py local-test
#minor differences in values and in some cases atom order
#adv-test not a known target
python2 scons/scons.py complete-test  
#all SUCCESS, again minor differences as before
python2 scons/scons.py pdb2pka-test
#fails, needs numpy
#use python_devirtualizer to install, this just puts all the required
#libraries in one place and then makes links to it.
pdv2ctrl mkvenv forpdb2pqr
pdv2ctrl install forpdb2pqr "numpy networkx"
pdv2ctrl migrate forpdb2pqr /tmp/forpdb2pqr
export PYTHONPATH=/tmp/forpdb2pqr/lib/python2.7/site-packages
ls $PYTHONPATH
  decorator-4.4.2.dist-info  easy_install.py         numpy                   pip-20.1.1.dist-info   setuptools                    _virtualenv.pth  wheel
decorator.py               networkx                numpy-1.16.6.dist-info  pip-20.1.1.virtualenv  setuptools-44.1.1.dist-info   _virtualenv.py   wheel-0.34.2.dist-info
decorator.pyc              networkx-2.2.dist-info  pip                     pkg_resources          setuptools-44.1.1.virtualenv  _virtualenv.pyc  wheel-0.34.2.virtualenv
python2 scons/scons.py pdb2pka-test
#fails, some problem with apbs not being set up for python
```
it was unhappy and emitted these messages:

```
Missing libraries for interfacing with APBS

You need to build APBS with Python support using the CMake variable -DENABLE_PYTHON=ON.

```
There is nothing in the dynamic libraries in apbs which appears to be python related:

```
ldd `which apbs` | grep -i py
#nothing
ldd `which apbs` | grep -i libapbs
        libapbs_mg.so => /usr/common/modules/el8/x86_64/software/apbs/1.5-CentOS-vanilla/lib/libapbs_mg.so (0x00007f6888c57000)
        libapbs_fem.so => /usr/common/modules/el8/x86_64/software/apbs/1.5-CentOS-vanilla/lib/libapbs_fem.so (0x00007f6888a44000)
        libapbs_routines.so => /usr/common/modules/el8/x86_64/software/apbs/1.5-CentOS-vanilla/lib/libapbs_routines.so (0x00007f688882b000)
        libapbs_generic.so => /usr/common/modules/el8/x86_64/software/apbs/1.5-CentOS-vanilla/lib/libapbs_generic.so (0x00007f68885f8000)
        libapbs_pmgc.so => /usr/common/modules/el8/x86_64/software/apbs/1.5-CentOS-vanilla/lib/libapbs_pmgc.so (0x00007f68883c9000)

```

In short, this is not installing/working at all well (on CentOS 8), and at least some of the problem is that it is still tied to the older python language.

**Comment by sobolevnrm** [comment]
All of PDB2PQR is now compatible with Python 3.
